conan/tools/cmake/cmakedeps/cmakedeps.py

```diff
@@ -11,6 +12,11 @@
 from conans.util.files import save
 
 
+FIND_MODE_MODULE = "module"
+FIND_MODE_CONFIG = "config"
+FIND_MODE_NONE = "none"
+FIND_MODE_BOTH = "both"
```

Comment:
I am curious why a `"both"` option? 
AFAIK official CMake are modules (And there are a few projects the provide their own) and most project install targets + follow https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html which are your config packages

but I've not come across a project that does both 🤔 with the old generators it was the costume who decided what to install but now it's defined by the recipe... we'll need to be more careful to define it correctly

Reply:
`both` cover the case where the library has a module in cmake, typical `FindXXX.cmake` but also the author's provided config (irrespective if we package it or not). In that case, you don't know how the consumer will expect to consume the package, maybe `find_package(XXX MODULE)` maybe `find_package(xxx CONFIG)` but we want Conan to be as "ready" as possible to consume any conan center package. So `CMakeDeps` will generate both. Note that the `config` namespace and target name should be declared typically following the author's one.

Reply:
That makes sense, I understand the motivation but I am not aware of any examples.

There are a few project out there that provide their own module files even though the upstream project provides config files. I wonder if this bonus feature will help there 🤔